### DISCUSSION TITLE
Kyber Ref Implementation Doubts - dead code ?

### DISCUSSION THREAD

**Discussion by abhi-dev-engg** [open-quantum-safe/liboqs#1756] [General]
Hello all,

I was analyzing Kyber768 ref code and have few doubts

1. Possible deadcode

Can this line be reached : https://github.com/open-quantum-safe/liboqs/blob/main/src/kem/kyber/pqcrystals-kyber_kyber512_ref/indcpa.c#L185 ?

GEN_MATRIX_NBLOCKS is 3 that makes **off** ( @ https://github.com/open-quantum-safe/liboqs/blob/main/src/kem/kyber/pqcrystals-kyber_kyber512_ref/indcpa.c#L184) always 0.

2. As per Kyber specification during encapsulation step , v is calculated as below 
v ← NTT−1(ˆt⊺ ◦ rˆ) +e2 + µ
However, i cannot seem to understand where is "t" transposed in the code :(

Please feel free to correct me incase I missed something ?

Thanks.

**Comment by SWilson4** [comment]
I suggest that you raise this with the [PQCrystals team](https://github.com/pq-crystals/kyber). We simply pull their implementation into `liboqs`, so they are better equipped to answer questions about details like this.

**Reply by abhi-dev-engg** [reply]
ok
